forge-game/src/main/java/forge/game/trigger/TriggerHandler.java

```diff
@@ -564,6 +564,12 @@ private void runSingleTriggerInternal(final Trigger regtrig, final Map<AbilityKe
             Card triggeredCard = ((Card) sa.getTriggeringObject(AbilityKey.Card));
             host.addRemembered(triggeredCard);
         }
+        if (regtrig.hasParam("LifePaid")) {
```

Comment:
Looks cleaner already :)

I think this check probably doesn't belong here, it's also a bit late and can cause more engine calls than needed
`meetsRequirementsOnTriggeredObjects` should be a better place for it

Reply:
Thanks – I'll try moving this